tests/test_workflow_init.py

```diff
@@ -241,6 +239,59 @@ def test_init_bids_symlink_mode(tmp_path):
     assert (dpath_root / "bids" / "README.md").exists()
 
 
+def test_init_bids_symlink_readonly(tmp_path):
+    """Create dummy BIDS dataset to use during init and use move symlink.
+
+    The BIDS dataset is read-only on disk w/o a README.
+    Make sure:
+    - InitWorkflow succeeds while warning no README could be created.
+    """
+    dpath_root = tmp_path / "nipoppy"
+    bids_to_link = tmp_path / "bids"
+    fids.create_fake_bids_dataset(
+        output_dir=bids_to_link,
+        subjects=["01"],
+        sessions=["1", "2"],
+        datatypes=["anat", "func"],
+    )
+
+    source_files_before_init = [
+        x.relative_to(bids_to_link) for x in bids_to_link.glob("**/*")
+    ]
+
+    # make the source is read-only
+    os.chmod(bids_to_link, 0o555)  # +rx
+
+    workflow = InitWorkflow(
+        dpath_root=dpath_root, bids_source=bids_to_link, mode="symlink"
+    )
+    workflow.run()
+
+    assert isinstance(workflow.manifest, Manifest)
+    assert workflow.manifest[Manifest.col_participant_id].to_list() == ["01", "01"]
+    assert workflow.manifest[Manifest.col_visit_id].to_list() == ["1", "2"]
+    assert workflow.manifest[Manifest.col_session_id].to_list() == ["1", "2"]
+    assert workflow.manifest[Manifest.col_datatype].to_list() == [
+        ["anat", "func"],
+        ["anat", "func"],
+    ]
+
+    source_files_after_init = [
+        x.relative_to(bids_to_link) for x in bids_to_link.glob("**/*")
+    ]
+
+    for f in source_files_before_init:
+        assert f in source_files_after_init
+
+    assert (dpath_root / "bids").is_symlink()
+    # only the directory is linked, not the files within
+
+    assert (dpath_root / "bids").readlink() == bids_to_link
+
+    assert len(source_files_after_init) == 24
```

Comment:
```suggestion
    assert len(source_files_after_init) == len(source_files_before_init)
```
To avoid hardcoding the number